lua/dap/session.lua

```diff
@@ -568,7 +568,12 @@ do
         if err1 then
           utils.notify('Error setting breakpoints: ' .. err1.message, vim.log.levels.ERROR)
         else
-          for _, bp in pairs(resp.breakpoints) do
+          for i, bp in pairs(resp.breakpoints) do
+            local payload_bp = payload.breakpoints[i]
+            if bp.line ~= payload_bp.line then
+                breakpoints.remove(bufnr, payload_bp.line, bp)
```

Comment:
I think this would discard the condition of conditional breakpoints or the message from log points